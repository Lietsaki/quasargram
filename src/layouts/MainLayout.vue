<template>
  <q-layout view="lHh Lpr lFf">
    <q-header class="bg-white text-grey-10 q-pa-xs" bordered>
      <q-toolbar class="constrain">
        <q-btn
          flat
          round
          color="primary"
          size="1.8rem"
          class="gt-xs text-grey-10 q-mr-sm"
          icon="eva-camera-outline"
          dense
          to="/camera"
        />

        <q-separator vertical inset spaced class="gt-xs" />
        <q-toolbar-title class="text-grand-hotel text-weight-bold">
          Quasargram
        </q-toolbar-title>
        <q-btn
          flat
          round
          color="primary"
          size="1.8rem"
          class="gt-xs text-grey-10"
          icon="eva-home-outline"
          dense
          to="/"
        />
      </q-toolbar>
    </q-header>

    <q-page-container class="bg-grey-1">
      <router-view />
    </q-page-container>

    <q-footer class="bg-white lt-sm" bordered>
      <q-tabs
        class="text-grey-10"
        active-color="primary"
        indicator-color="transparent"
      >
        <q-route-tab to="/" icon="eva-home-outline" />
        <q-route-tab to="/camera" icon="eva-camera-outline" />
      </q-tabs>
    </q-footer>
  </q-layout>
</template>

<script>
export default {
  name: 'MainLayout',
  components: {},
  data() {
    return {
      leftDrawerOpen: false
    };
  }
};
</script>

<style lang="scss">
// .q-toolbar {
//   @media (min-width: $breakpoint-sm-min) {
//     height: 7rem;
//   }
// }

.q-toolbar__title {
  font-size: 30px;

  @media (max-width: $breakpoint-xs-max) {
    text-align: center;
  }
}

.q-footer {
  .q-tab__icon {
    width: 30px;
    height: 30px;
    font-size: 30px;
  }
}
</style>
