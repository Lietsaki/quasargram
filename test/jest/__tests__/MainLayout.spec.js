import { mountQuasar } from '@quasar/quasar-app-extension-testing-unit-jest';
import {
  QLayout,
  QHeader,
  QToolbar,
  QBtn,
  QSeparator,
  QToolbarTitle,
  QPageContainer,
  QFooter,
  QTabs,
  QRouteTab
} from 'quasar';
import MainLayout from '../../../src/layouts/MainLayout';

const mountQuasarConfig = {
  mount: {
    stubs: ['router-link', 'router-view']
  },
  quasar: {
    components: {
      QLayout,
      QHeader,
      QToolbar,
      QBtn,
      QSeparator,
      QToolbarTitle,
      QPageContainer,
      QFooter,
      QTabs,
      QRouteTab
    }
  }
};

describe('Main Layout', () => {
  test('Displays title correctly', () => {
    // 1) Mount the MainLayout component
    const wrapper = mountQuasar(MainLayout, mountQuasarConfig);

    // 2) Get the element with a class of .text-grand-hotel
    const appTitle = wrapper.find('.text-grand-hotel');

    // 3) Expect such element to contain the 'Quasargram' text
    expect(appTitle.text()).toContain('Quasargram');

    // 4) Take a snapshot and expect it to match the last one
    expect(appTitle).toMatchSnapshot();
  });
});
