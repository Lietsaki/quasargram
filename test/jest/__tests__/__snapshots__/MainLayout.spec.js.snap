// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Layout Displays title correctly 1`] = `
<q-toolbar-title-stub class="text-grand-hotel text-weight-bold">
  Quasargram
</q-toolbar-title-stub>
`;
